centraldogma/src/main/java/com/linecorp/decaton/centraldogma/CentralDogmaPropertySupplier.java

```diff
@@ -145,19 +148,43 @@ public static CentralDogmaPropertySupplier register(CentralDogma centralDogma, S
         return new CentralDogmaPropertySupplier(centralDogma, project, repository, filename);
     }
 
+    public static CentralDogmaPropertySupplier register(CentralDogma centralDogma, String project,
+                                                        String repository, String filename,
+                                                        PropertySupplier supplier) {
+        List<Property<?>> properties = defaultProperties().stream().map(defaultProperty -> {
+            if (supplier.getProperty(defaultProperty.definition()).isPresent()) {
+                return supplier.getProperty(defaultProperty.definition()).get();
```

Comment:
As a call of `getProperty` may involves I/O with external systems (like centraldogma), it is better to keep count of calling it as minimum as possible.
so let's do it like:

```java
prop = supplier.getProperty(...);
if (prop.ifPresent()) {
    return prop.get();
}
...
```

Reply:
I use a temporary variable for keeping the value in external systems. https://github.com/line/decaton/pull/134/commits/8d38efefa231e638194e922b2f1a37ded6641185